libkbfs/block_journal.go

```diff
@@ -762,28 +718,33 @@ func (j *blockJournal) removeFlushedEntry(ctx context.Context,
 		return err
 	}
 
-	// TODO: This is a hack to work around KBFS-1439. Figure out a
-	// better way to update j.refs to reflect the removed entry.
-	refs, err := j.readJournal(ctx)
-	if err != nil {
-		return err
+	for id, idContexts := range entry.Contexts {
+		refs := j.refs[id]
+		if len(refs) == 0 {
+			continue
+		}
+		for _, context := range idContexts {
+			err := refs.remove(context, earliestOrdinal)
```

Comment:
Under what scenario would the tag not match `earliestOrdinal`?  How could it ever be different? The block ids and refnonces are both effectively random and couldn't be put in two different journal entries, as far as I can tell. Is this just a safety check?  It would be nice to have some comments here explaining what's going on.